### PR TITLE
Add sonar configuration in platform

### PR DESCRIPTION
**Pull request by edouardbe**

### CODE REVIEW

File: itm/itm_test.go [hunk 1]
```diff
@@ -159,3 +182,37 @@ func TestUserAgentStringOnRequest(t *testing.T) {
...
+			// Compare non-map values
+			if expectedValue != receivedValue {
+				return false // Values not equal
+
+			}
+		}
+	}
+
+	return true
+}
```
Comment (tolgaio) [inline itm/itm_test.go:218]: how about using a library for this https://github.com/google/go-cmp

```
if !cmp.Equal(receivedValue, expectedValue) {
    t.Errorf("Wrong object received, got=%s", cmp.Diff(expectedValue, receivedValue))
}
```

File: itm/dns_apps_test.go [hunk 2]
```diff
@@ -51,7 +28,7 @@ func TestErrorIssuingPostOnCreateDNSApps(t *testing.T) {
 	if app != nil {
 		t.Error("Expected nil result")
 	}
-	expectedError := "Post https://itm.cloud.com:443/api/v2/config/applications/dns.json?publish=false: foo"
+	expectedError := "Post \"https://itm.cloud.com:443/api/v2/config/applications/dns.json?publish=false\": foo"
```
Comment (tolgaio) [inline itm/dns_apps_test.go:31]: nit: you can use backtick here for readability

```
`Post "https://itm.cloud.com:443/api/v2/config/applications/dns.json?publish=false"`
```

Comment (tolga-mubi) [inline itm/dns_apps_test.go]: nit: you can use backtick here for readability

```
`Post "https://itm.cloud.com:443/api/v2/config/applications/dns.json?publish=false"`
```

File: itm/itm_test.go (hunk 1 above)
Comment (tolga-mubi) [inline itm/itm_test.go:218]: how about using a library for this https://github.com/google/go-cmp

```
if !cmp.Equal(receivedValue, expectedValue) {
    t.Errorf("Wrong object received, got=%s", cmp.Diff(expectedValue, receivedValue))
}
```

**Review by tolga-mubi — APPROVED** [review]

File: itm/itm_test.go (hunk 1 above)
Comment (edouardbe) [reply itm/itm_test.go:218]: In the caller,
if !compareMaps(expectedRequestData, parsedBody) {
			t.Error(unexpectedValueString("Request body", expectedRequestData, parsedBody))
		}

the parsedBody has more entries than the expectedRequestData. 

the compareMaps returns true only if all the entries in expectedRequestData are at least in parsedBody.

the cmp.Equal expects that every entries are in both maps.

File: itm/dns_apps_test.go (hunk 2 above)
Comment (edouardbe) [reply itm/dns_apps_test.go]: I will change, thanks for the tips

**Comment by edouardbe** [comment]
@tolga-mubi , I force pushed on the first commit to remove the \" . If you are fine with the compareMaps I did, I will rebase and merge. No squash, they are independent commits.

**Pull request merged**